K2Bridge/Visitors/Aggregations/TermsAggregationVisitor.cs

```diff
@@ -21,25 +20,39 @@ public void Visit(TermsAggregation termsAggregation)
             EnsureClause.StringIsNotNullOrEmpty(termsAggregation.Metric, nameof(TermsAggregation.Metric));
             EnsureClause.StringIsNotNullOrEmpty(termsAggregation.Field, nameof(TermsAggregation.Field));
 
-            var query = new StringBuilder();
+            // Extend expression: ['2']=['Carrier']
+            var extendExpression = $"{EncodeKustoField(termsAggregation.Key)} = {EncodeKustoField(termsAggregation.Field, true)}";
 
-            // Add main aggregation query (summarize)
-            // KQL ==> _data | summarize ['key1']=metric(field1), ['key2']=metric(field2), count() by ['key']=field
-            query.Append($"{KustoTableNames.Data} | {KustoQLOperators.Summarize} {termsAggregation.SubAggregationsKustoQL}{termsAggregation.Metric} by {EncodeKustoField(termsAggregation.Key)} = {EncodeKustoField(termsAggregation.Field, true)}");
+            // Bucket expression: count() by ['2']=['Carrier'] | order by count_ desc | limit 5
+            var bucketExpression = new StringBuilder();
+            bucketExpression.Append($"{termsAggregation.Metric} by {EncodeKustoField(termsAggregation.Key)}");
 
             var orderBy = termsAggregation.Order?.SortField switch
             {
-                "_key" => $"{KustoQLOperators.CommandSeparator}{KustoQLOperators.OrderBy} {EncodeKustoField(termsAggregation.Key)} {termsAggregation.Order.SortOrder}",
-                "_count" => $"{KustoQLOperators.CommandSeparator}{KustoQLOperators.OrderBy} {BucketColumnNames.Count} {termsAggregation.Order.SortOrder}",
-                { } s => $"{KustoQLOperators.CommandSeparator}{KustoQLOperators.OrderBy} {EncodeKustoField(s)} {termsAggregation.Order.SortOrder}",
+                "_key" => $"{KustoQLOperators.CommandSeparator} {KustoQLOperators.OrderBy} {EncodeKustoField(termsAggregation.Key)} {termsAggregation.Order.SortOrder}",
+                "_count" => $"{KustoQLOperators.CommandSeparator} {KustoQLOperators.OrderBy} {EncodeKustoField(BucketColumnNames.Count)} {termsAggregation.Order.SortOrder}",
```

Comment:
We don't really need to encode fields we know ahead of time

Reply:
Yes, I agree this is not really needed. But I just considered it offers clarity in the KQL produced, everything is encoded properly.